From a905c5bee2758b4968455700bfd58e1286c0f597 Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Wed, 8 Feb 2023 16:38:25 +0800
Subject: [PATCH 23/85] fix tsan for hnswlib and hnswfast

---
 contrib/faiss   | 2 +-
 contrib/hnswlib | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/contrib/faiss b/contrib/faiss
index e36f4c1fd0..c679689cf1 160000
--- a/contrib/faiss
+++ b/contrib/faiss
@@ -1 +1 @@
-Subproject commit e36f4c1fd0b1f57f4721fc29d16df62ed5d31920
+Subproject commit c679689cf10eff6043b1333f2d8b764746dd9e7c
diff --git a/contrib/hnswlib b/contrib/hnswlib
index 7659d94583..90fe3a1119 160000
--- a/contrib/hnswlib
+++ b/contrib/hnswlib
@@ -1 +1 @@
-Subproject commit 7659d94583bf9df015a7de9d4e7377a800210f33
+Subproject commit 90fe3a11192af08f0207a354a5b0e3a9fd177f2c
-- 
2.32.1 (Apple Git-133)

