From 667ec92e1236a1336521b477b7c3b05af35cdd3a Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Tue, 4 Jul 2023 10:50:46 +0800
Subject: [PATCH 7/7] update search-index

---
 contrib/search-index | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/search-index b/contrib/search-index
index 0673de049f..d3a4b45d89 160000
--- a/contrib/search-index
+++ b/contrib/search-index
@@ -1 +1 @@
-Subproject commit 0673de049f95b2a20c86ac0bff019c0c8d87fe30
+Subproject commit d3a4b45d893a86b47b6c315f3a0da600acaa2ef6
-- 
2.32.1 (Apple Git-133)

