From f7c60610315679c804f393bd8fb03f22f933e9cc Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Tue, 13 Dec 2022 14:17:54 +0800
Subject: [PATCH 013/150] remove unused definitions in compilation

---
 scripts/config_on_linux.sh | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/scripts/config_on_linux.sh b/scripts/config_on_linux.sh
index 0f71f9c9f1..a44d206be3 100644
--- a/scripts/config_on_linux.sh
+++ b/scripts/config_on_linux.sh
@@ -21,11 +21,8 @@ cmake -G Ninja .. -DCMAKE_C_COMPILER=$(command -v clang-13) \
     -DENABLE_S3=ON \
     -DENABLE_SSL=ON \
     -DENABLE_REPLXX=ON \
-    -DENABLE_ARROW=ON \
     -DUSE_UNWIND=ON \
     -DENABLE_UTILS=OFF \
     -DENABLE_TESTS=OFF \
     -DENABLE_RAPIDJSON=ON \
-    -DENABLE_LICENSE_CHECK=OFF \
-    -DVERSION_GITHASH="111"
-
+    -DENABLE_LICENSE_CHECK=OFF
-- 
2.32.1 (Apple Git-133)

