From 68853579d2c06a2d7c3309e6b18e815b119c76e6 Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Fri, 28 Apr 2023 13:53:48 +0800
Subject: [PATCH 03/49] fix sanitizer compilation of search-index

---
 contrib/search-index | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/search-index b/contrib/search-index
index 38415f2271..f6169681df 160000
--- a/contrib/search-index
+++ b/contrib/search-index
@@ -1 +1 @@
-Subproject commit 38415f2271b66c5c793aa04a962684df7f430fc0
+Subproject commit f6169681df197a3ad57f9ea21d7bfd54f58b0f0f
-- 
2.32.1 (Apple Git-133)

