From 1daba137331a11ab72e5431123498f8c565721ab Mon Sep 17 00:00:00 2001
From: Shanfeng <shanfengp@moqi.ai>
Date: Tue, 9 May 2023 15:54:19 +0800
Subject: [PATCH 10/49] fix undefined behavior bug in logf

---
 base/glibc-compatibility/musl/logf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/glibc-compatibility/musl/logf.c b/base/glibc-compatibility/musl/logf.c
index 7ee5d7fe62..e4c2237caa 100644
--- a/base/glibc-compatibility/musl/logf.c
+++ b/base/glibc-compatibility/musl/logf.c
@@ -53,7 +53,7 @@ float logf(float x)
 	tmp = ix - OFF;
 	i = (tmp >> (23 - LOGF_TABLE_BITS)) % N;
 	k = (int32_t)tmp >> 23; /* arithmetic shift */
-	iz = ix - (tmp & 0x1ff << 23);
+	iz = ix - (tmp & 0xff800000);
 	invc = T[i].invc;
 	logc = T[i].logc;
 	z = (double_t)asfloat(iz);
-- 
2.32.1 (Apple Git-133)

