From 0e34ea658dc6010e234dfee7615cf26f4b21c914 Mon Sep 17 00:00:00 2001
From: shanfengp <shanfengp@moqi.ai>
Date: Tue, 16 May 2023 10:44:57 +0800
Subject: [PATCH 17/49] skip annoy test case

---
 tests/queries/0_stateless/02354_annoy.sql | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/queries/0_stateless/02354_annoy.sql b/tests/queries/0_stateless/02354_annoy.sql
index 6a0ff60179..98d637ef86 100755
--- a/tests/queries/0_stateless/02354_annoy.sql
+++ b/tests/queries/0_stateless/02354_annoy.sql
@@ -1,4 +1,4 @@
--- Tags: no-fasttest, no-ubsan, no-cpu-aarch64
+-- Tags: no-fasttest, no-ubsan, no-cpu-aarch64, no-msan
 
 SET allow_experimental_annoy_index = 1;
 
-- 
2.32.1 (Apple Git-133)

