From 638d9744170532b07ad1500a23c7b2cc69d40b15 Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Tue, 16 May 2023 12:03:21 +0800
Subject: [PATCH 19/49] update search-index for filter search

---
 contrib/search-index | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/search-index b/contrib/search-index
index b939424a46..6be787cfa6 160000
--- a/contrib/search-index
+++ b/contrib/search-index
@@ -1 +1 @@
-Subproject commit b939424a46913018e5781d396e01a9cd31582243
+Subproject commit 6be787cfa6a101ec27e02dfb16ed8cf996fcef26
-- 
2.32.1 (Apple Git-133)

