From b96472883471c0b3529275b13dc33ccf47f901ae Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Fri, 26 May 2023 18:55:28 +0800
Subject: [PATCH 31/49] fix filter search error of MSTG

fix #151
---
 contrib/search-index | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/search-index b/contrib/search-index
index c028d113f9..7b50734984 160000
--- a/contrib/search-index
+++ b/contrib/search-index
@@ -1 +1 @@
-Subproject commit c028d113f9cff34123b9918141357345d7995dfc
+Subproject commit 7b50734984c2271da3cbd2734ff3cb1b80614ec4
-- 
2.32.1 (Apple Git-133)

