From 295970b1c809c0a0ace4d99a1d23d0efe1976a05 Mon Sep 17 00:00:00 2001
From: Qin Liu <lqgy2001@gmail.com>
Date: Mon, 29 May 2023 23:24:20 +0800
Subject: [PATCH 33/49] update search-index

---
 contrib/search-index | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/search-index b/contrib/search-index
index 7b50734984..bea7a83da2 160000
--- a/contrib/search-index
+++ b/contrib/search-index
@@ -1 +1 @@
-Subproject commit 7b50734984c2271da3cbd2734ff3cb1b80614ec4
+Subproject commit bea7a83da2e16cd5f42dfd28c140eab08d7e77aa
-- 
2.32.1 (Apple Git-133)

